daffodil-lib/src/main/scala/edu/illinois/ncsa/daffodil/util/DecimalUtils.scala

```diff
@@ -0,0 +1,342 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package edu.illinois.ncsa.daffodil.util
+
+import edu.illinois.ncsa.daffodil.schema.annotation.props.gen.BinaryNumberCheckPolicy
+
+import java.math.{ BigInteger => JBigInteger, BigDecimal => JBigDecimal }
+
+object DecimalUtils {
+
+  def signCodesToHex(signCodes: String, policy: BinaryNumberCheckPolicy): Map[String, List[Int]] = {
+    // remove all spaces from the string
+    val str: String = signCodes.replaceAll("\\s", "")
+    val chars: Array[Char] = str.toCharArray()
+
+    val valid_positives: List[Int] = List(
+      Integer.parseInt(String.valueOf('A'), 16),
+      Integer.parseInt(String.valueOf('C'), 16),
+      Integer.parseInt(String.valueOf('E'), 16),
+      Integer.parseInt(String.valueOf('F'), 16))
+    val valid_negatives: List[Int] = List(
+      Integer.parseInt(String.valueOf('B'), 16),
+      Integer.parseInt(String.valueOf('D'), 16))
+    val valid_unsigneds: List[Int] = List(
+      Integer.parseInt(String.valueOf('F'), 16))
+    val valid_zero_signs: List[Int] = List(
+      Integer.parseInt(String.valueOf('A'), 16),
+      Integer.parseInt(String.valueOf('C'), 16),
+      Integer.parseInt(String.valueOf('E'), 16),
+      Integer.parseInt(String.valueOf('F'), 16),
+      Integer.parseInt(String.valueOf('0'), 16))
+
+    val positives: List[Int] = List(Integer.parseInt(String.valueOf(chars(0)), 16))
+    val negatives: List[Int] = List(Integer.parseInt(String.valueOf(chars(1)), 16))
+    val unsigneds: List[Int] = List(Integer.parseInt(String.valueOf(chars(2)), 16))
+    val zero_signs: List[Int] = List(Integer.parseInt(String.valueOf(chars(3)), 16))
```

Comment:
Also something to think about with this stuff, if someone had dfdl:binaryPackedSignCodes="C D F 0" but dfdl:binaryNumberCheckPolicy="lax", parsing should accept all the valid values, but unparsing should output C D F and 0. I think the code you have will just pick the first thing in the list, which would be A B F A. So either ordering matters when building these lists, or unparsing should have an instance different from parsing with just the dfdl:binaryPackedSignCodes and ignoreing the valid stuff. Make sure to add tests for this, where it could parse one thing but unparse a different thing